specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/preview/2019-02-01-preview/imagebuilder.json

```diff
@@ -0,0 +1,1148 @@
+            "$ref": "./examples/UpdateImageTemplateTags.json"
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "VirtualMachineImageTemplate"
+        ],
+        "operationId": "VirtualMachineImageTemplate_Get",
+        "description": "Get information about a virtual machine image template",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ImageTemplateNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ImageTemplate"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ApiError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Retrieve an Image Template.": {
+            "$ref": "./examples/GetImageTemplate.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "VirtualMachineImageTemplate"
+        ],
+        "operationId": "VirtualMachineImageTemplate_Delete",
+        "description": "Delete a virtual machine image template",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ImageTemplateNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation was successful."
+          },
+          "202": {
+            "description": "The operation will be completed asynchronously."
+          },
+          "204": {
+            "description": "NoContent -- VM image template does not exist in the subscription."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ApiError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via":"azure-async-operation"
+        },
+        "x-ms-examples": {
+          "Delete an Image Template.": {
+            "$ref": "./examples/DeleteImageTemplate.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.VirtualMachineImages/imageTemplates/{imageTemplateName}/run": {
```

Comment:
is there no scenario where the user would need to PUT the image and run it in a template deployment? Given that this 1) creates a resource, and 2) may be useful to execute in a template, it seems like implementing this as:
`PUT /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.VirtualMachineImages/imageTemplates/{imageTemplateName}/runOutputs/{runOutputName}` is a better pattern

Reply:
We have that as an idea for a future version. It will need a more thorough redesign though and we don't want to block these features on that.

Reply:
It would be optimal to resolve this before introducing a stable API. Thanks for the update